ros/src/foxglove_bridge/src/generic_client.cpp

```diff
@@ -59,36 +59,60 @@ std::string getTypeIntrospectionSymbolName(const std::string& serviceType) {
          pkgName + "__" + (middleModule.empty() ? "srv" : middleModule) + "__" + typeName;
 }
 
-/**
- * The default symbol names for getting type support handles for services are missing from the
- * rosidl_typesupport_cpp shared libraries, see
- * https://github.com/ros2/rosidl_typesupport/issues/122
- *
- * We can however, as a (hacky) workaround, use other symbols defined in the shared library.
- * With `nm -C -D /opt/ros/humble/lib/libtest_msgs__rosidl_typesupport_cpp.so` we see that there is
- * `rosidl_service_type_support_t const*
- * rosidl_typesupport_cpp::get_service_type_support_handle<test_msgs::srv::BasicTypes>()` which
- * mangled becomes
- * `_ZN22rosidl_typesupport_cpp31get_service_type_support_handleIN9test_msgs3srv10BasicTypesEEEPK29rosidl_service_type_support_tv`
- * This is the same for galactic, humble and rolling (tested with gcc / clang)
- *
- * This function produces the mangled symbol name for a given service type.
- *
- * \param[in] serviceType The service type, e.g. "test_msgs/srv/BasicTypes"
- * \return Symbol name for getting the service type support handle
- */
-std::string getServiceTypeSupportHandleSymbolName(const std::string& serviceType) {
+const rosidl_service_type_support_t* GenericClient::getServiceTypeSupportHandle(
+  const std::string& serviceType) {
+#if RCLCPP_VERSION_GTE(25, 0, 0)
+  // Jazzy and newer can use the built-in rclcpp call.
+  return rclcpp::get_service_typesupport_handle(serviceType, TYPESUPPORT_LIB_NAME,
+                                                *_typeSupportLib);
+#else
+  // Humble needs to do additional work.
   const auto [pkgName, middleModule, typeName] = extract_type_identifier(serviceType);
+
+  // Humble rosidl 3.1.6 is the first version to have the appropriate
+  // __get_service_type_support_handle__ methods available.  However, rosidl does not have a version
+  // macro, so we approximate this with rclcpp versions.
+  // https://github.com/ros2/ros2/commit/9f7d334291ff77a07d68d4d8976871b99e03612b is the version
+  // where rosidl was bumped to 3.1.6, and also bumped rclcpp to version 16.0.11.  Thus, anything
+  // older than that version needs to use the old symbol hack.
```

Comment:
I feel like we shouldn't include the old symbol hack anymore. It is probably fine to assume that the humble rosidl version includes the `__get_service_type_support_handle__` methods. If people still want to build the bridge for older humble versions, they can use the non-SDK based version at https://github.com/foxglove/ros-foxglove-bridge

Reply:
All right, removed that compatibility bit in cc86c6e.